x-pack/plugins/maps/public/classes/sources/join_sources/types.ts

```diff
@@ -41,3 +40,7 @@ export interface ITermJoinSource extends ISource {
   ): Promise<ITooltipProperty[]>;
   getFieldByName(fieldName: string): IField | null;
 }
+
+export interface ITermJoinSource extends IJoinSource {
+  getTermField(): IField;
+}
```

Comment:
nit, might be nice to export a type checking function that can replace the existing (and possible future) instances of `'getTermField' in (joinRightSource as ITermJoinSource)` elsewhere in our code ([here](https://github.com/elastic/kibana/pull/155575/files#diff-b112d6df5191ff5f4f3096ff465cd0fc5955edbb32f4708a455a5b431dd0f1cfR87) and [here](https://github.com/elastic/kibana/pull/155575/files#diff-3c61668ab2a265d2e928a792f6d1e6a855c99d949e0b0c772cc6d7932cf3f258R1000)).